test/weiner_convolve_test.cc

```diff
@@ -0,0 +1,378 @@
+                                              make_tuple(16, 17, func),
+                                              make_tuple(32, 17, func)};
+        return ::testing::ValuesIn(params);
+    }
+};
+
+class AV1WienerConvolveHbdTest
+    : public AV1WienerConvolveTest<uint16_t, HbdWienerConvolveFunc,
+                                   HbdWienerConvolveParam> {
+  protected:
+    void run_check_output(const int test_times) {
+        out_w_ = TEST_GET_PARAM(0);
+        out_h_ = TEST_GET_PARAM(1);
+        func_tst_ = TEST_GET_PARAM(2);
+        bd_ = TEST_GET_PARAM(3);
+        // Generate random filter kernels
+        DECLARE_ALIGNED(16, InterpKernel, hkernel);
+        DECLARE_ALIGNED(16, InterpKernel, vkernel);
+
+        uint8_t* input = CONVERT_TO_BYTEPTR(input_);
+        uint8_t* out_tst = CONVERT_TO_BYTEPTR(output_tst_);
+        uint8_t* out_ref = CONVERT_TO_BYTEPTR(output_ref_);
+        const ConvolveParams conv_params = get_conv_params_wiener(bd_);
+        for (int kernel_type = 0; kernel_type < 3; kernel_type++) {
+            generate_kernels(hkernel, vkernel, kernel_type);
+            for (int i = 0; i < test_times; ++i) {
+                for (int i = 0; i < h; ++i)
+                    for (int j = 0; j < w; ++j)
+                        input_[i * w + j] = rnd16_.random() & ((1 << bd_) - 1);
+                // Choose random locations within the source block
+                int offset_r = 3 + pseudo_uniform(h - out_h_ - 7);
+                int offset_c = 3 + pseudo_uniform(w - out_w_ - 7);
+                av1_highbd_wiener_convolve_add_src_c(
+                    input + offset_r * w + offset_c,
+                    w,
+                    out_ref,
+                    out_w_,
+                    hkernel,
+                    16,
+                    vkernel,
+                    16,
+                    out_w_,
+                    out_h_,
+                    &conv_params,
+                    bd_);
+                func_tst_(input + offset_r * w + offset_c,
+                          w,
+                          out_tst,
+                          out_w_,
+                          hkernel,
+                          16,
+                          vkernel,
+                          16,
+                          out_w_,
+                          out_h_,
+                          &conv_params,
+                          bd_);
+
+                for (uint32_t j = 0; j < out_w_ * out_h_; ++j)
+                    ASSERT_EQ(output_tst_[j], output_ref_[j])
+                        << "Pixel mismatch at index " << j << " = ("
+                        << (j % out_w_) << ", " << (j / out_w_)
+                        << ") on iteration " << i;
+            }
+        }
+    }
+
+  public:
+    static ::testing::internal::ParamGenerator<HbdWienerConvolveParam>
+    BuildParams(HbdWienerConvolveFunc func) {
+        const HbdWienerConvolveParam params[] = {make_tuple(8, 8, func, 8),
```

Comment:
Why the test size in hbd is much smaller than lbd?

Reply:
don't know, copied from reference